features/nanostack/mbed-mesh-api/source/WisunBorderRouter.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "WisunInterface.h"
+#include "ns_trace.h"
+#include "WisunBorderRouter.h"
+#include "MeshInterfaceNanostack.h"
+
+extern "C" {
+#include "ws_bbr_api.h"
+}
+
+#define TRACE_GROUP "WSBR"
+
+mesh_error_t WisunBorderRouter::start(NetworkInterface *mesh_if, NetworkInterface *backbone_if)
+{
+    InterfaceNanostack *nano_mesh_if = reinterpret_cast<InterfaceNanostack *>(mesh_if);
+    int8_t mesh_if_id = nano_mesh_if->get_interface_id();
+    if (mesh_if_id < 0) {
+        return MESH_ERROR_UNKNOWN;
+    }
+    _mesh_if_id = mesh_if_id;
+
+    char backbone_if_name[7] = {0};
+    if (backbone_if->get_interface_name(backbone_if_name) == NULL) {
+        return MESH_ERROR_UNKNOWN;
+    }
+
+    if (strlen((char *) &backbone_if_name) < 4) {
+        return MESH_ERROR_UNKNOWN;
+    }
+
+    int backbone_if_id = atoi(&backbone_if_name[3]);
```

Comment:
Can `get_interface_name` be renamed and extended to return also if_id? Or maybe separate method for it?

Reply:
Interface id is a nanostack specific feature, so I think we should not add high level NetworkInterface.h (or EMACInterface.h) methods for that, if we can avoid it. 

We might need at some point to change for backhaul from NetworkInterface::get_default_instance() to more specific nanostack backhaul interface constructors, but a the moment the default instance is working as it is.